// react-ocl/lib/components/SmilesSvgRenderer.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import React, { memo } from 'react';
import { useHandleMemoError } from '../hooks/useHandleMemoError';
import { ErrorRenderer, DefaultErrorRenderer } from './ErrorRenderer';
import SvgRenderer from './SvgRenderer';
function SmilesSvgRenderer(props) {
  const {
    OCL,
    smiles,
    ErrorComponent = DefaultSmilesErrorComponent,
    ...otherProps
  } = props;
  const [error, mol] = useHandleMemoError(() => OCL.Molecule.fromSmiles(smiles), [OCL, smiles]);
  if (error) {
    return /*#__PURE__*/React.createElement(ErrorRenderer, {
      width: props.width,
      height: props.height,
      ErrorComponent: ErrorComponent,
      value: smiles,
      error: error
    });
  }
  return /*#__PURE__*/React.createElement(SvgRenderer, _extends({
    mol: mol
  }, otherProps));
}
export default /*#__PURE__*/memo(SmilesSvgRenderer);
function DefaultSmilesErrorComponent(props) {
  return /*#__PURE__*/React.createElement(DefaultErrorRenderer, {
    height: props.height,
    message: "Invalid SMILES"
  });
}
